### DISCUSSION TITLE
Governance of FastAPI, Typer et al.

### DISCUSSION THREAD

**Discussion by followben** [fastapi/fastapi#4738] [Questions] [answered]
Hi folks,

First to say @tiangolo has done (and continues to do) an amazing job building FastAPI and associated tools like Typer, and the documentation is generally excellent.

~One thing that does seem to be lacking is any discussion around the governance of both of these projects~. I'm not an active contributor, but in using and recommending these tools professionally, I'd like to understand the development process incl. who has the power to make architectural decisions and merge urgent PRs. Having a BDFL is great, but most successful open source projects of this scale also have a dedicated team of core committers who ensure the project keeps moving forward.

It does concern me there are over 900 issues and almost 500 PRs awaiting review for FastAPI alone (compare those stats to alternatives such as Flask). As we double down on using these projects in building critical infrastructure, I want to be confident urgent fixes are reviewed and merged promptly and all users and stakeholders have clear understanding of the future roadmap.

Thoughts?

Best,
Ben.

EDIT: @tiangolo has already attempted to address this question at https://twitter.com/tiangolo/status/1403336990121148422 and https://github.com/tiangolo/fastapi/discussions/3970#discussioncomment-1927924.

**Comment by tiangolo** [comment, marked as answer]
Yep, about the number of open issues, most are actually questions, it's unfortunate that the word is "issues". But now that GitHub Discussions have an API and also they finally have templates, I was able to update the code to compute FastAPI People and start recommending questions in GitHub Discussions.

Now, the problem and the bottleneck is not actually being able to read issues (anyone can read them) nor even hitting the "merge" button... the bottleneck in issues and questions is actually sitting to understand the questions and issues, answer them, creating self-contained examples, etc. And the bottleneck in PRs is properly reviewing them, but not just hitting the "approve" button, anyone can do that and there's no special permission required to do it, but actually following all the process to review the PR and make sure that it really works and really solves the problem.

I would really like and welcome all the help I can get, unfortunately, there's just a bunch of people coming to really help, most come only to ask, but anyway.

If you want to help, you can answer questions: https://fastapi.tiangolo.com/help-fastapi/#help-others-with-questions-in-github and you can review PRs: https://fastapi.tiangolo.com/help-fastapi/#review-pull-requests
